lib/vscode/src/vs/server/node/server.ts

```diff
@@ -209,7 +209,7 @@ export class Vscode {
 	// ../../electron-browser/sharedProcess/sharedProcessMain.ts#L148
 	// ../../../code/electron-main/app.ts
 	private async initializeServices(args: NativeParsedArgs): Promise<void> {
-		const environmentService = new NativeEnvironmentService(args);
+		const environmentService = new NativeEnvironmentService(args, this.services.get<IProductService>(IProductService) as IProductService);
```

Comment:
The product service hasn't been set at this point so I'm thinking we can do:

```
const productService = { _serviceBrand: undefined, ...product };
```

Then use that here as well as in the `set` call below (line 247).